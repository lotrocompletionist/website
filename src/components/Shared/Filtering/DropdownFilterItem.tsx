import * as React from "react";
import { DropdownItem } from "bloomer";

export interface DropdownFilterItemProps {
  id: string;
  value: string;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
}

export function DropdownCheckboxFilterItem(props: DropdownFilterItemProps) {
  return (
    <DropdownItem>
      <input
        id={props.id}
        className="is-checkradio"
        type="checkbox"
        value={props.value}
        onChange={props.onChange}
        name={props.value}
      />
      <label htmlFor={props.id}>{props.value}</label>
    </DropdownItem>
  );
}
