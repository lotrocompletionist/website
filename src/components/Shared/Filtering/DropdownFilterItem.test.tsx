import * as React from "react";
import { create } from "react-test-renderer";
import { DropdownCheckboxFilterItem } from "./DropdownFilterItem";

test("<DropdownCheckboxFilterItem> renders correctly", () => {
  const id = "region-id";
  const value = "Ered-Luin";
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {};
  const tree = create(
    <DropdownCheckboxFilterItem id={id} value={value} onChange={onChange} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
