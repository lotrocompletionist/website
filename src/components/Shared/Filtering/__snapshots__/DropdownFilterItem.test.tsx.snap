// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownCheckboxFilterItem> renders correctly 1`] = `
<div
  className="dropdown-item"
>
  <input
    className="is-checkradio"
    id="region-id"
    name="Ered-Luin"
    onChange={[Function]}
    type="checkbox"
    value="Ered-Luin"
  />
  <label
    htmlFor="region-id"
  >
    Ered-Luin
  </label>
</div>
`;
